### PR TITLE
[WIP] initial check in for ktica

### PR DESCRIPTION
**Pull request by msultan**
This is basically @evanfeinberg 's implementation of ktica using the nystorem landmark approximation for kernalizing the input data. I basically added a wrapper for the nystorem and another for the pipeline. Still need to add tests, docs, and the option to add custom landmarks.

### CODE REVIEW

**Comment by mpharrigan** [comment]
noice

**Comment by rmcgibbo** [comment]
I'll let others make the call, but I'm -1. The file ``msmbuilder/decomposition/kernal_approx.py`` seems like a good addition -- +1 for that. But the other file doesn't add anything new. It's just a lot of complicated docstrings around a very simple `Pipeline`.

**Comment by mpharrigan** [comment]
I'll take a closer look later

File: msmbuilder/decomposition/ktica.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+        Returns
+        -------
+        sequence_new : list of array-like, each of shape (n_samples_i, n_components)
+
+        """
+        check_iter_of_sequences(sequences, max_iter=3)  # we might be lazy-loading
+
+        X_transformed = self._kt_pipeline.transform(sequences)
+        #n_x = self.nystorm.transform(X)
```
Comment (cxhernandez) [inline msmbuilder/decomposition/ktica.py]: Why are these here?

Comment (msultan) [inline msmbuilder/decomposition/ktica.py]: I am updating the docs. gimme a few hours, I was trying to add tests first.

**Comment by msultan** [comment]
@all we could potentially just add a command line script and move the rest of the code to somewhere else. Any recommendations?

**Comment by cxhernandez** [comment]
Yeah. Maybe make an example of this for the docs?

**Comment by evanfeinberg** [comment]
Thanks, @msultan, for integrating this. It should be pretty straightforward for us to add custom landmark points. Please see this modified Nystroem class:

https://github.com/evanfeinberg/conformation/blob/master/LandmarkNystroem.py

So if we just import this instead of the standard Nystroem, the feature is available. If you don't pass the argument "basis", it generates a random set of basis points/landmarks, or you can pass a list of basis points.

**Comment by mpharrigan** [comment]
@evanfeinberg have you considered proposing that change upstream to sklearn

**Comment by msultan** [comment]
yea i agree with @mpharrigan's suggestion. Though if they dont want it, we can always have a local variant too.

**Comment by mpharrigan** [comment]
Ok, I agree that we shouldn't do a weird pipeline thing to combine two steps

@msultan : do you want to re-work this pr and just include the nystroem wrapper?
